www/scipylib/building/windows.rst

```diff
@@ -32,185 +32,257 @@ information than this document will provide.
+
+You should now have all of the tools required to build OpenBLAS. Now change directories
+to your favorite location with the following:
+ 
+.. code:: shell
+
+   cd /c/Users/MyUser/Downloads/
+   
+To make sure that we're ready to build, type the following in the terminal:
+
+.. code:: shell
+
+   make
+   gfortran
+   gcc
+
+You don't necessarily need to build in that particular location, but it should be somewhere
+reasonable. Now clone the repository required to build OpenBLAS:
+
+.. code:: shell
+
+   git clone https://github.com/matthew-brett/build-openblas.git
+   cd build-openblas
+   git submodule update --init --recursive
+
+If any of these commands fail, you're not ready to build. Go back and make sure that MSYS2
+is installed correctly and has the required packages enabled. Now, let's set some
+environment variables. In the MSYS2 terminal, type the following.
+
+.. code:: shell
+
+    export OPENBLAS_COMMIT=5f998ef
+    export OPENBLAS_ROOT="C:\\opt"
+    export BUILD_BITS=64
+
+Please check these variables' purpose for a moment. More specifically, make sure that
+you have  write/delete access to the path :code:`OPENBLAS_ROOT` points to. The output of the
+OpenBLAS build will  be collected in this folder. Make sure that the :code:`OPENBLAS_COMMIT`
+points to the correct OpenBLAS commit that you want to build in the cloned repo. In the
+future, :code:`build_openblas` repository might get updated and you might want to get those
+updates by changing the commit. Make sure that the architecture is correctly set to either
+32 or 64 bit. And after you've made sure of that, start the OpenBLAS build with:
+
+.. code:: shell
+
+    ./build_openblas.sh
+
+Building OpenBLAS is extremely problematic and may fail if your system is not correctly
+configured. Your build may fail after a few hours and you may have to restart it after 
+fixing an undocumented problem. OpenBLAS builds can also fail silently and produce an
+incorrect binary. Please, if you have any issues, `report them`_ so that we can save the
+next person's time.
+
+While you're waiting on OpenBLAS to finish building, go ahead and install `build tools`_
+from Microsoft, since these take a while to install and you'll need them later.
+
+After the :code:`build_openblas.sh` script has completed (probably with an error), there
+should be an :code:`openblas.a` file somewhere on your system. If :code:`OPENBLAS_ROOT` was
+set to :code:`C:\\opt`, then you might see a line like this in the MSYS2 terminal:
+
+.. code:: shell
+
+   Copying the static library to /c/opt/64/lib
+
+Installing OpenBLAS
+===================
+
+If you see that line, then you might have OpenBLAS correctly, even if other failures
+occurred. Look in that folder for :code:`openblas.a`. If you find a file called something
+like :code:`libopenblas_5f998ef_gcc7_2_0.a`, just rename it to :code:`openblas.a` and continue.
+If the file isn't there, then poke around and try to find the file elsewhere. If you don't
```

Comment:
The file can't be arbitrarily placed right? If it is not in C:\opt, then `make .... install` has failed and the files didn't show up there. So this makes it a bit mysterious as if file can be placed somewhere that we have to figure out.